packages/guides-restructured-text/src/RestructuredText/Directives/ContentsDirective.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\RestructuredText\Directives;
+
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\Nodes\TocNode;
+use phpDocumentor\Guides\RestructuredText\Parser\Directive;
+use phpDocumentor\Guides\RestructuredText\Parser\DocumentParserContext;
+use phpDocumentor\Guides\UrlGeneratorInterface;
+
+/**
+ * Standarad rst `contents` directive
+ *
+ * Displays a table of content of the current page
+ */
+class ContentsDirective extends BaseDirective
+{
+    public function __construct(private readonly UrlGeneratorInterface $urlGenerator)
+    {
+    }
+
+    public function getName(): string
+    {
+        return 'contents';
+    }
+
+    /** {@inheritDoc} */
+    public function process(
+        DocumentParserContext $documentParserContext,
+        Directive $directive,
+    ): Node|null {
+        $options = $directive->getOptions();
+        $absoluteUrl = $this->urlGenerator->absoluteUrl(
+            $documentParserContext->getContext()->getDirName(),
+            $documentParserContext->getContext()->getCurrentFileName(),
+        );
+
+        return (new TocNode([$absoluteUrl]))
```

Comment:
Should this be a TocNode or a contentnode that you introduced in this pr? 

If you use the contentnode, you might need to adjust the compiler pass so it accepts the content node. 
It might be worth it to us the contents node so people can manipulate the TOC and contents directive result differently.

Reply:
Yeah, I was thinking about this as well. But is there such a concept of toc / contents outside of rst?

Reply:
If we introduce a contents node, one could use it to create a compilerpass to create a content node based on the heading of a document. Even if the markup language doesn't support it.

There are some extensions on markdown to support a table of contents, like sphinx does for RST. As the original format was not designed to have such feature. 

The commonmark library we are planning to use for markdown has such extension.

Reply:
I gave it another try to unify the menu handling